signac_dashboard/modules/navigator.py

```diff
@@ -0,0 +1,99 @@
+from flask import render_template, url_for
+
+from signac_dashboard.module import Module
+from signac_dashboard.util import escape_truncated_values
+
+
+class Navigator(Module):
+    """Displays links to jobs differing in one state point parameter.
+
+    This module may not update if the signac project changes because the module
+    caches the project schema when the signac-dashboard launches.
+
+    :param context: Supports :code:`'JobContext`
+    :type context: str
+    """
+
+    _supported_contexts = {"JobContext"}
+
+    def __init__(
+        self,
+        name="Navigator",
+        context="JobContext",
+        template="cards/navigator.html",
+        **kwargs,
+    ):
+        super().__init__(name=name, context=context, template=template, **kwargs)
+
+    def get_cards(self, job):
+        project = self._dashboard.project
+
+        def link_label(key, other_val):
+            """Returns the url and label for the job with job.sp[key] == other_val."""
+
+            similar_sp = job.sp()  # modifiable
+            similar_sp.update({key: other_val})
+
+            # Look only for exact matches of only changing one parameter
+            # in case of heterogeneous schema
+            other_job = project.open_job(similar_sp)
+            if other_job in project:
+                link = url_for("show_job", jobid=other_job.id)
+                label = other_job.sp[key]
+            else:
+                link = None
+                label = "no match"
+            return link, label
+
+        nearby_jobs = dict()
+        # for each parameter in the schema, find the next and previous job and get links to them
+        for key, values in self._sorted_schema.items():
+            my_val = job.sp.get(key, None)
```

Comment:
Do you need a special placeholder value here? Note that `None` is a legal value in a state point, and we should support that value rather than making it mean "key not present." We sometimes use an instance of a placeholder class instead of `None` for this exact purpose. Example: https://github.com/glotzerlab/signac/blob/4d5f8a57bd25430c00364d9556b1de7be1ace91b/signac/_search_indexer.py#L45-L46

Reply:
I was wondering about this as well. I'll use this